### PR TITLE
Add the BERT-based scalar weighting model

### PR DESCRIPTION
**Pull request by danieldk**
Note: k in top-k is always 3. I have created an issue #10 for making this configurable. I didn't want to change this in this PR, since it requires changes throughout various parts of `sticker` and it would make the `master` branch and PRs diverge too much from the `wip` branch.

### CODE REVIEW

File: sticker/src/model/mod.rs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+#[derive(Debug)]
+pub struct BertModel {
+    embeddings: BertEmbeddings,
+    encoder: BertEncoder,
+    classifiers: HashMap<String, ScalarWeightClassifier>,
+    layers_dropout: Dropout,
+}
+
+impl BertModel {
+    /// Construct a model and load parameters.
```
Comment (twuebi) [inline sticker/src/model/mod.rs]: Which parameters are loaded?

Comment (danieldk) [reply sticker/src/model/mod.rs]: None ;).

File: sticker/src/model/mod.rs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+    /// Do a forward pass of the model on the inputs.
+    ///
+    /// * `attention_mask`: specifies which sequence elements should
+    ///    be masked when applying the encoder.
+    /// * `train`: indicates whether this forward pass will be used
+    ///   for backpropagation.
+    /// * `freeze_embeddings`: exclude embeddings from backpropagation.
+    /// * `freeze_encoder`: exclude the encoder from backpropagation.
+    pub fn forward_t(
```
Comment (twuebi) [inline sticker/src/model/mod.rs]: is `forward_t` used for training?

Comment (danieldk) [reply sticker/src/model/mod.rs]: It's currently not used at all, so I have removed it from this PR.

File: sticker/src/model/mod.rs [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        Ok(BertModel {
+            embeddings,
+            encoder,
+            layers_dropout: Dropout::new(layers_dropout),
+            classifiers,
+        })
+    }
+
+    /// Encode an input.
+    fn encode(
```
Comment (twuebi) [inline sticker/src/model/mod.rs:121]: Wouldn't `forward_t` be the fitting name for this?

Comment (danieldk) [reply sticker/src/model/mod.rs:121]: But `encode` only produces hidden states. I would expect a `forward_t` method to also apply the scalar weighting and perhaps softmax.

Comment (twuebi) [reply sticker/src/model/mod.rs:121]: Ah I see. Still a bit undecided though: from a `BertModel::forward_t` I'd expect the bert encoder and not scalar weighted classification.

**Review by twuebi — APPROVED** [review]
